test/runnable/mars1.d

```diff
@@ -2180,6 +2180,30 @@ void test21038()
     assert((cast(size_t) name21038) % dchar.sizeof == 0);
 }
 
+////////////////////////////////////////////////////////////////////////
+// https://issues.dlang.org/show_bug.cgi?id=16274
+
+extern(C) int pair(short a, ushort b, byte c, ubyte d);
+
+struct S
+{
+    // provide alternate implementation of .pair()
+    pragma(mangle, "pair")
+    extern(C) static void pair(int a, int b, int c, int d)
```

Comment:
wouldn't it be much better to implement this on the C side, to test the C-ABI interop? 
The point of this change is not to force arguments to become `int`, it's to match C ABI, which is slightly different (imagine a platform where the C ABI is _not_ to promote to `int`).

Reply:
and explicitly test this with some extreme values where things may go wrong (2 and 4 don't sound very difficult)

Reply:
> wouldn't it be much better to implement this on the C side, to test the C-ABI interop?

1. I couldn't get gcc to access those upper bits.
2. Tests are better self-contained as much as practical.
3. There are plenty of other tests done in C that check short and int arguments.
4. Any unpredictable C ABI changes are going to cause unpredictable problems, experience shows it's a waste of time attempting to predict them. The wackiest one is the Win64 one which took a left turn at Alberqueue into bizzaro land. Nobody could have predicted that.

Reply:
Yeah, this test totally won't pass on ARM, SPARC, or POWER (to name three widely supported alternative

Reply:
> I couldn't get gcc to access those upper bits.

The original implementation of init_pair is here:
https://github.com/mirror/ncurses/blob/790a85dbd4a81d5f5d8dd02a44d84f01512ef443/ncurses/base/lib_color.c#L556-L703

Though as the host platform in the bug report is OSX, I suspect it is clang that is compiling the C code.

Reply:
> 4. Any unpredictable C ABI changes are going to cause unpredictable problems, experience shows it's a waste of time attempting to predict them.

Isn't that the point of testcases though? To detect when things no longer work and a fix is needed?

Reply:
If D's `extern (C)` isn't working, not much is going to work. Here, we don't need to test that `extern (C)` is working.